content/docs/1.10.0/advanced-resources/rebuilding/_index.md

```diff
@@ -1,4 +1,113 @@
+
+### Delta Replica Rebuilding
+
+- This is available for failed replica reusage only, and there is an existing snapshot file (with the same name) in the failed replica data directory
+- There is no checksum/checksum file for the snapshot file. It's like a fallback of fast rebuilding
+
+- **Pros**:
+  - Reduce network bandwidth consumption.
+- **Cons**:
+  - Increased CPU overhead.
+  - Rebuilding time is influenced by CPU performance.
+
+### Fast Replica Rebuilding
+
+Fast rebuilding is enabled when:
+
+- The fast replica building setting is enabled:
+  - `fast-replica-rebuild-enabled: true`
+- Snapshot checksum files are created:
+  - `snapshot-data-integrity`: `enabled` or
+  - `snapshot-data-integrity-immediate-check-after-snapshot-creation: true`
+
+6. Skip checking unchanged snapshots if the checksums are the same in the checksum files of snapshots.
+7. If the value of snapshot checksum files between the target replica and the source replica does not exist or is not identical, it will go to the delta rebuilding flow.
+
+- **Pros**:
+  - Minimize network bandwidth consumption.
+  - Minimize disk IO.
+- **Cons**:
+  - Calculating the snapshot checksum can be time-consuming.
+  - The timing of checksum calculation is unpredictable. It can be triggered even if the node is experiencing heavy IO load.
+
+The snapshot checksum is computed:
+
+- Immediately after the snapshot is created with the `snapshot-data-integrity-immediate-check-after-snapshot-creation` setting enabled.
+- Regularly (by default: 7 days) if the `snapshot-data-integrity` setting is `enabled`.
+
+For more details, [Fast Replica Rebuilding](./fast-replica-rebuilding).
+
+## Factors That Affect Rebuilding Performance
+
+- Large volume head or No snapshot exists
+  - **WHY**:  
+    - If the replica fails and rebuilding starts immediately, there is no opportunity to generate a checksum file for this large volume head. Then Longhorn has to do a full rebuild for all data in this large volume head
+    - If the replica soonly fails and gets rebuilt after the snapshot of this large volume head is created, there is not enough time to calculate the checksum. As a result, Longhorn has to do a delta rebuilding for all data in this newly created snapshot, which will cause checksum calculation overhead and numerous checksum comparisons between the target replica and the source replica
+  - **HOW**:
+    1. Enable `snapshot-data-integrity-immediate-check-after-snapshot-creation` or `snapshot-data-integrity` setting, so checksums are precomputed.
+    2. Take regularly snapshots by a recurring job.
+- Snapshot Purged
+  - **WHY**:  
+  When the snapshot purged of volume starts, the system-generated snapshot will be coalesced to the next snapshot, and the checksum of the next snapshot will be invalid.
+  - **HOW**:
+    1. Enable `snapshot-data-integrity-immediate-check-after-snapshot-creation` to ensure the checksum precomputed after the snapshot purged.
```

Comment:
Please elaborate the trade-off the setting `snapshot-data-integrity-immediate-check-after-snapshot-creation`.

Reply:
Added a section to analyze the setting trade-off. Thanks.

Reply:
Please briefly mention the trade-off here as well. it will be more clear for users.

Reply:
Added. Thanks.